Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/ElidingLabel.cpp

```diff
@@ -39,8 +42,24 @@ namespace AzQtComponents
         m_metricsLabel->setText(m_text);
         
         m_elidedText.clear();
-        elide();
-        updateGeometry();
+
+        if (QThread::currentThread() != QApplication::instance()->thread())
+        {
+            // If setText is called from a thread other than the main thread,
+            // fontMetrics().elidedText() will result in a crash.
+            QTimer::singleShot(
+                0,
+                [&]()
+                {
+                    elide();
+                    updateGeometry();
+                });
+        }
+        else
+        {
+            elide();
+            updateGeometry();
+        }
```

Comment:
Is there a QTBUG that is a known issue for this being called from a different thread? What is the backtrace?

We should move this check to the top of the method so you can just do a singleShot timer that calls `setText` instead of duplicating logic.

Reply:
you can use a signal and slot and qt should queue it if its on another thread. this seems to be hiding a bug if your going to do this

Reply:
I can't see a QTBUG for this, there are a couple of other asserts when setting the tooltip that complain about sending signals across threads, but they are skipable and removing the code that causes them does not fix the crash.

Reply:
I've changed the code so that it always runs the elide in a timer. It's tidier and won't hurt existing code. The thread check wasn't strictly necessary so I've removed it.

Reply:
@pollend I've changed the code to use a signal rather than the timer and I've moved the m_elidedText.clear() to inside the threadsafe section. Setting m_metricsLabel should have no effect as it's a QLabel which doesn't exhibit any issues when used across threads.